tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -61,6 +84,16 @@ def test_use_params(self):
     with self.assertRaises(Exception):
       self.scheduler.product_request(str, [Params(a)])
 
+  def test_transitive_params(self):
```

Comment:
Could you add a comments on the two scheduler invokes in this test? I'm not quite following how it tests the case described in the commit message.